stm32-modules/heater-shaker/firmware/motor_task/mc_api.c

```diff
@@ -1,463 +1,463 @@
- * that are currently active.
- *
- * See \link Fault_generation_error_codes Motor Control Faults\endlink for a list of
- * of all possible faults codes.
- */
-__weak uint16_t MC_GetCurrentFaultsMotor1(void)
-{
-	return MCI_GetCurrentFaults( pMCI[M1] );
-}
-
-/**
- * @brief returns the current state of Motor 1 state machine
- */
-__weak State_t  MC_GetSTMStateMotor1(void)
-{
-	return MCI_GetSTMState( pMCI[M1] );
-}
-
+/**
+  ******************************************************************************
+  * @file    mc_api.c
+  * @author  Motor Control SDK Team, ST Microelectronics
+  * @brief   This file implements the high level interface of the Motor Control SDK.
+  ******************************************************************************
+  * @attention
+  *
+  * <h2><center>&copy; Copyright (c) 2019 STMicroelectronics.
+  * All rights reserved.</center></h2>
+  *
+  * This software component is licensed by ST under Ultimate Liberty license
+  * SLA0044, the "License"; You may not use this file except in compliance with
+  * the License. You may obtain a copy of the License at:
+  *                             www.st.com/SLA0044
+  *
+  ******************************************************************************
+  */
+
+#include "mc_interface.h"
+#include "mc_api.h"
+#include "mc_config.h"
+
+/** @addtogroup MCSDK
+  * @{
+  */
+
+/** @defgroup MCIAPI Motor Control API
+  *
+  * @brief High level Programming Interface of the Motor Control SDK
+  *
+  *  This interface allows for performing basic operations on the motor(s) driven by an
+  * Motor Control SDK based application. With it, motors can be started and stopped, speed or
+  * torque ramps can be programmed and executed and information on the state of the motors can
+  * be retrieved, among others.
+  *
+  *  This interface consists in functions that target a specific motor, indicated in their name.
+  * These functions aims at being the main interface used by an Application to control motors.
+  *
+  *  The current Motor Control API can cope with up to 2 motors.
+  * @{
+  */
+
+extern MCI_Handle_t * pMCI[NBR_OF_MOTORS];
+
+/**
+  * @brief  Initiates the start-up procedure for Motor 1
+  *
+  * If the state machine of Motor 1 is in #IDLE state, the command is immediately
+  * executed. Otherwise the command is discarded. The Application can check the
+  * return value to know whether the command was executed or discarded.
+  *
+  * One of the following commands must be executed before calling MC_StartMotor1():
+  *
+  * - MC_ProgramSpeedRampMotor1()
+  * - MC_ProgramTorqueRampMotor1()
+  * - MC_SetCurrentReferenceMotor1()
+  *
+  * Failing to do so results in an unpredictable behaviour.
+  *
+  * @note The MC_StartMotor1() command only triggers the start-up procedure:
+  * It moves Motor 1's state machine from the #IDLE to the #IDLE_START state and then
+  * returns. It is not blocking the application until the motor is indeed running.
```

Comment:
to be clear i didn't write this, this is generated code. but what it means is that the two ways stuff like this works is either
1. what they didn't do:
```
void MC_StartMotor1() {
   start(); // do some stuff to kick off the background controller tasks
   while(!started()) {}
   // by the time this function returns, the motor is spinning
}
```
2. what they did do
```
void MC_StartMotor1() {
   start(); // do some stuff to kick off the background controller tasks
  // it's on the higher levels of the application to do whatever scheduling they want and figure out when the motor is actually spinning, this just returns now
}
```

If they'd done 1, and this function didn't return, the thread that called it would be just spinning waiting for the motor, unable to do anything else like handle messages or anything. Since they've done 2, there could be a little more effort required, but you're at least able to do other stuff from the calling task.